providers/gce/OWNERS

```diff
@@ -1,6 +1,7 @@
 # See the OWNERS docs at https://go.k8s.io/owners
-
-approvers:
+# We are no longer accepting features into k8s.io/legacy-cloud-providers.
```

Comment:
Maybe we do not need these statements in out-of-tree providers.

Reply:
Let's also add @jiahuif to reviewers (in addition to you and I).

Reply:
Updated and added @jiahuif